File: turnstile-backend/src/main/java/it/tdgc/turnstile/util/ApiResponse.java
package it.tdgc.turnstile.util;

import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Getter
@Setter
public class ApiResponse<T> {
    private String status;
    private String message;
    private T data;
    private Date timestamp;
    private Object metadata;

    public ApiResponse(String status, String message, T data, Date timestamp,  Object metadata) {
        this.status = status;
        this.message = message;
        this.data = data;
        this.timestamp = timestamp;
        this.metadata = metadata;
    }


}

File: turnstile-frontend/src/app/models/api-response.ts
export  class ApiResponse<T> {
    status : string ;
    message: string;
    data : T[];
    timestamp: string;
    metadata : Object;
    constructor(data: T[], status: string, message: string, timestamp: string, metadata: Object) {
        this.data = data;
        this.status = status;
        this.message = message;
        this.timestamp = timestamp;
        this.metadata = metadata;
    }
}
